jena-text-es/src/main/java/org/apache/jena/query/text/es/TextIndexES.java

```diff
@@ -422,6 +422,27 @@ public EntityDefinition getDocDef() {
     }
 
     private String parse(String fieldName, String qs, String lang) {
+        //Escape special characters if any in the query string
+        qs = qs.replaceAll("\\:", "\\\\:")
```

Comment:
This may be a question of style, but might this read a bit better as a constant `Pattern`? (I don't know enough to think very usefully about the performance implications, but I would think a `Pattern` would be at _least_ as performant.)

Reply:
Awesome. I will wait it to be merged in master.